solutions/cf-588E.mdx

```diff
@@ -0,0 +1,168 @@
+---
+id: cf-588E
+source: CF
+title: Duff in the Army
+author: Chongtian Ma
+---
+
+## Explanation
+
+Note that for each two nodes $u$ and $v$, the path between them is just $u$ to
+$lca(u,v)$ and $v$ to $lca(u,v)$. Also note the constraint of $a \leq 10$ for
+each query. To answer each query, we only need to store the $10$ people with the
+smallest ids for the path from $u$ to $lca(u,v)$ and $v$ to $lca(u,v)$, and
+combine them. This can be done with binary lifting, which will store the $10$
+smallest ids for each vertex to their $2^k$'th ancestor.
+
+## Implementation
+
+We can answer each query in $\mathcal{O}(a \log n)$. Even though the constant
+factor is pretty bad, the problem limits give room for it. Note
+that if you do store more than $10$ smallest ids the problem will yield a MLE!
+
+**Time Complexity**: $\mathcal{O}(a \log n)$ for each query
```

Comment:
```suggestion
**Time Complexity**: $\mathcal{O}(Q \times A \log N)$
```

Reply:
I'm fine with it as is. if you want the full time complexity it would also need to include $na\log n + m\log m$